kubemarine/procedures/upgrade.py

```diff
@@ -118,13 +118,14 @@ def upgrade_containerd(cluster: KubernetesCluster):
     if cri == 'containerd':
         path = 'plugins."io.containerd.grpc.v1.cri"'
         target_kubernetes_version = cluster.context["upgrade_version"]
-        index_pos = target_kubernetes_version.rfind(".")
-        target_kubernetes_version = target_kubernetes_version[:index_pos]
-        pause_version = cluster.globals['compatibility_map']['software']['pause'][target_kubernetes_version]['version']
+        pause_mapping = cluster.globals['compatibility_map']['software']['pause']
+        if target_kubernetes_version not in pause_mapping:
+            raise Exception(f"Upgrade to {target_kubernetes_version} is not supported")
+        pause_version = pause_mapping[target_kubernetes_version]['version']
         if not cluster.inventory["services"]["cri"]['containerdConfig'].get(path, False):
             return
         last_pause_version = cluster.inventory["services"]["cri"]['containerdConfig'][path]["sandbox_image"].split(":")[2]
-        if last_pause_version != pause_version:
+        if True:
             sandbox = cluster.inventory["services"]["cri"]['containerdConfig'][path]["sandbox_image"]
```

Comment:
Previously `pause_version` was specified in globals as 3.7 and was of **float** type.  `last_pause_version` is of *str* type. *float* was always != *str* so the condition was always true. Now `pause_version` is specified in globals as '3.7' and becomes always == `last_pause_version`, because `last_pause_version` is already enriched from globals. This is actually a big gap in the pause upgrade procedure.

Reply:
tracked MANOPD-87623